### ISSUE TITLE
new HA update freezes

### ISSUE THREAD

**Issue by max-rh** [peterbuga/HASS-sonoff-ewelink#97]
Hi Peter,

Excellent work with the last light update; thanks alot for that :).
I am running HA version 0.91.4 and started freezing every few min, i am checking the logs and i get a few sonoff errors that i think causing the issue but i am not sure; thought i could update you with that.

Here are the errors i keep getting:
`error from callback <bound method Sonoff.on_message of <custom_components.sonoff.Sonoff object at 0x7f01e01fce10>>: local variable 'state' referenced before assignment

error from callback <bound method Sonoff.on_message of <custom_components.sonoff.Sonoff object at 0x7f01e01fce10>>: 'NoneType' object has no attribute 'attributes'

error from callback <bound method Sonoff.on_message of <custom_components.sonoff.Sonoff object at 0x7f01e01fce10>>: can only concatenate str (not "int") to str`

Thanks

**Comment by peterbuga** [comment]
hi @max-rh please correct me if i'm wrong but i assume you're using the latest `sonoff-bulb` dev branch, i mean this is the only one that makes sense to throw the above errors, if so, i just pushed a fix for it (and i really hope it'll work because i have no way to test it right now 😬)

i do assume those errors were triggered when you updated a device from the eWeLink app, can you please confirm this **before** updating to the latest fix i just push? also, it could help me if you could list-mention the device types you have in your account (i'm under the impression you already did this before but i don't remember where and there's a possibility you added more of them meanwhile)

ps: another thing that'll help me is if you copy-paste the full stack of an error. you can usually find them if you watch the HA log in command line as they give far more information of where the problem actually starts/started.

thanks

**Comment by max-rh** [comment]
Hi Peter,
Thanks alot for replying; I think i was using the one before the latest sonoff-bulb version, as for the ewelink update, i am not sure, i did update them using the app but i am not sure of the timeline if it was at the same time the error appeared.

Will update you with the full logs tomorrow hopefully; I am running HA within a docker container so i need to access the container to get the full logs.

**Comment by peterbuga** [comment]
@max-rh no more freezes 😅 ?

**Comment by max-rh** [comment]
haven't push the latest sonoff-bulb update yet, but looks like no more freezes till now, i guess its as u said; i think it was because of ewelink update while the component is loaded.

Thanks Peter, you can go ahead and close this issue, sorry been busy lately, but will send you bunch of logs hopefully to be helpful to u; and may i suggest to add an option to disable the component, or atleast to turn of the polling rate; as you did with the debug button, this will be helpful to just hit a button in HA then update the devices through the app.

Thanks alot peter; great work as always.

**Comment by peterbuga** [comment]
no worries, when you have the time there's no rush.
unfortunately, i cannot just disable the polling it's an important fallback to update/get the current state of devices. with the new changes (because websockets are handling most of the tasks) i hard increased the polling period to a minimum of 60sec and i tend to assume is quite decent vs the 10 seconds on the earlier implementations.

there might be also a possibility that more and more people started using this component (or parts of this code) and basically I/we created a small ddos network that is constantly requesting data from itead servers 😬 🙈 
bottom line is that in the end i'll try to debug and improve this component as best as I can!

**Comment by max-rh** [comment]
Hi Peter,
thought that i could update you with the latest logs; i got a few freezes last week, had to reboot HA to solve it; i noticed that once HA looses internet connectivity, something happens, but i am not sure if this is the issue...I have my own DNS server and a do a lot of changes on it so i keep rebooting it; hence HA looses connectivity for couple of min sometimes. 
 
and i am also getting the same "Sonoff.on_message of <custom_components.sonoff.Sonoff object at 0x72351230>>: 'mappingproxy' object has no attribute 'attributes'" that was mentiones in the other issue @jkoenig72 opened.

I am not sure if this because of my DNS, cause it blocks alot of my requests; tried looking up whats the sonoff server to whitlist but with no luck, would appreciated it if you could provide me with what i should whitlist to make the connection valid.

[sonoff.txt](https://github.com/peterbuga/HASS-sonoff-ewelink/files/3175055/sonoff.txt)

**Comment by peterbuga** [comment]
@max-rh check/test this if you want https://github.com/peterbuga/HASS-sonoff-ewelink/commit/b587974d91e7ad5064ebe160a3ea794f0e9709b4 (the change is here https://github.com/peterbuga/HASS-sonoff-ewelink/commit/b587974d91e7ad5064ebe160a3ea794f0e9709b4#diff-a1b8ad5ad0f6fb61d3538acc6b861c95R359)

**Comment by peterbuga** [comment]
following this issue https://github.com/peterbuga/HASS-sonoff-ewelink/issues/154 and some further testings the component works ok/stable on latest HA 0.99.3 raspbian and/or hassio
